edx_solutions_api_integration/courses/views.py

```diff
@@ -1832,6 +1832,141 @@ class CoursesMetrics(SecureAPIView):
+            org_ids=[org_id] if org_id else None
+        )
+        cache_course_data(cache_category, course_id, {'course_avg': avg_grade})
+
+        return avg_grade
+
+    @staticmethod
+    def get_course_avg_progress(course_id, org_id=None, percent_completion=True):
+        """
+        Get average progress score of a course
+        If org_id is passed then score is limited to that org's users
+        """
+        cache_category = 'progress'
+        if org_id:
+            cache_category = '{}_{}'.format(cache_category, org_id)
+
+        data = get_cached_data(cache_category, course_id)
+        if data is not None:
+            return data.get('course_avg')
+
+        course_key = get_course_key(course_id)
+        exclude_user_ids = get_aggregate_exclusion_user_ids(course_key)
+
+        data = _get_course_progress_metrics(
+            course_key,
+            exclude_users=exclude_user_ids,
+            org_ids=[org_id] if org_id else None,
+            percent_completion=percent_completion,
+        )
+        cache_course_data(cache_category, course_id, data)
+
+        return data.get('course_avg')
+
+    @staticmethod
+    def get_course_passed_users_count(course_id, org_id=None):
+        """
+        Get passed users count of a course
+        if org_id is passed then count is limited to that org's users
+        """
+        cache_category = 'passed_count'
+        if org_id:
+            cache_category = '{}_{}'.format(cache_category, org_id)
+
+        passed_count = get_cached_data(cache_category, course_id)
+        if passed_count is not None:
+            return passed_count
+
+        course_key = get_course_key(course_id)
+        exclude_user_ids = get_aggregate_exclusion_user_ids(course_key)
+
+        passed_count = StudentGradebook.get_passed_users_gradebook(
+            course_key,
+            exclude_users=exclude_user_ids,
+            org_ids=[org_id] if org_id else None
+        ).count()
+        cache_course_data(cache_category, course_id, passed_count)
+
+        return passed_count
+
+    @staticmethod
+    def get_course_completed_users_count(course_id, org_id=None):
+        """
+        Get completed users count of a course
+        if org_id is passed then count is limited to that org's users
+        """
+        cache_category = 'completed_count'
+        if org_id:
+            cache_category = '{}_{}'.format(cache_category, org_id)
+
+        completed_count = get_cached_data(cache_category, course_id)
+        if completed_count is not None:
+            return completed_count
+
+        course_key = get_course_key(course_id)
+        exclude_user_ids = get_aggregate_exclusion_user_ids(course_key)
+
+        completed_count = StudentGradebook.get_num_users_completed(
+            course_key,
+            exclude_users=exclude_user_ids,
+            org_ids=[org_id] if org_id else None
+        )
+        cache_course_data(cache_category, course_id, completed_count)
```

Comment:
Should we invalidate it or is 30 minutes cache is fine?

Reply:
I think for aggregated scores, it should be fine it they delay by 30 minutes